qiskit_ibm/ibm_provider.py

```diff
@@ -902,19 +789,21 @@ def services(self) -> Dict:
         """Return all available services.
 
         Returns:
-            All services available to this provider.
+            All services available to this account.
         """
         return {key: val for key, val in self._services.items() if val is not None}
 
-    def __eq__(
+    def _get_hgp_by_service_and_backend_name(
```

Comment:
Can this be combined with `_get_hgp()`, with service/backend name as filters? It just feels a bit chaotic to have 3 different methods that all filter and return `hgp`